client/tailscale/localclient.go

```diff
@@ -259,6 +259,22 @@ func (lc *LocalClient) DaemonMetrics(ctx context.Context) ([]byte, error) {
 	return lc.get200(ctx, "/localapi/v0/metrics")
 }
 
+// ClientMetric is the request data for UploadClientMetric.
+type ClientMetric struct {
+	Name string `json:"name"`
+	// One of "counter" or "gauge"
+	Type  string `json:"type"`
+	Value int    `json:"value"`
+}
+
+// UploadClientMetrics posts metric to the Tailscale daemon.
+// If any metric doesn't exist yet, the daemon creates a new one with the given
+// name and initial value.
+func (lc *LocalClient) UploadClientMetrics(ctx context.Context, metrics []ClientMetric) error {
```

Comment:
The slice should only contain one item per Name, right?

Reply:
Based on the localapi implementation, it doesn't really have to - if included multiple times, the metric value will be increased both of those times.